test/e2e-tests.sh

```diff
@@ -0,0 +1,317 @@
+    popd
+  fi
+
+  # Wait For All The Knative Eventing Namespace To Be Removed
+  wait_until_object_does_not_exist namespaces knative-eventing
+}
+
+# Install Strimzi Kafka Operator & Cluster Into K8S Cluster
+function kafka_setup() {
+
+  # Create The Namespace Where Strimzi Kafka Will Be Installed
+  echo "Installing Kafka Cluster"
+  kubectl create namespace ${STRIMZI_KAFKA_NAMESPACE} || return 1
+
+  # Install Strimzi Into The Desired Namespace (Dynamically Changing The Namespace)
+  sed "s/namespace: .*/namespace: ${STRIMZI_KAFKA_NAMESPACE}/" ${STRIMZI_CLUSTER_OPERATOR_CONFIG} | kubectl apply -f - -n ${STRIMZI_KAFKA_NAMESPACE}
+
+  # Wait For The Strimzi Kafka Cluster Operator To Be Ready (Forcing Delay To Ensure CRDs Are Installed To Prevent Race Condition)
+  wait_until_pods_running ${STRIMZI_KAFKA_NAMESPACE} || fail_test "Failed to start up Strimzi Cluster Operator"
+
+  # Create The Actual Kafka Cluster Instance For The Cluster Operator To Setup
+  kubectl apply -f ${STRIMZI_KAFKA_CLUSTER_CONFIG} -n ${STRIMZI_KAFKA_NAMESPACE}
+
+  # Delay Pod Running Check Until All Pods Are Created To Prevent Race Condition (Strimzi Kafka Instance Can Take A Bit To Spin Up)
+  local iterations=0
+  local progress="Waiting for Kafka Pods to be created..."
+  while [[ $(kubectl get pods --no-headers=true -n ${STRIMZI_KAFKA_NAMESPACE} | wc -l) -lt 6 && $iterations -lt 60 ]] # 1 ClusterOperator, 3 Zookeeper, 1 Kafka, 1 EntityOperator
+  do
+    echo -ne "${progress}\r"
+    progress="${progress}."
+    iterations=$((iterations + 1))
+    sleep 3
+  done
+  echo "${progress}"
+
+  # Wait For The Strimzi Kafka Cluster To Be Ready
+  wait_until_pods_running ${STRIMZI_KAFKA_NAMESPACE} || fail_test "Failed to start up a Strimzi Kafka Instance"
+}
+
+# Uninstall Strimzi Kafka Operator & Cluster From K8S Cluster
+function kafka_teardown() {
+  echo "Uninstalling Kafka Cluster"
+  kubectl delete -f ${STRIMZI_KAFKA_CLUSTER_CONFIG} -n ${STRIMZI_KAFKA_NAMESPACE}
+  kubectl delete -f ${STRIMZI_CLUSTER_OPERATOR_CONFIG} -n ${STRIMZI_KAFKA_NAMESPACE}
+  kubectl delete namespace ${STRIMZI_KAFKA_NAMESPACE}
+}
+
+# Install The eventing-kafka KafkaChannel Implementation Via Ko
+function channel_setup() {
+
+  # Copy The eventing-kafka Config To Temp Directory For Modification
+  echo "Installing KafkaChannel"
+  cp ${EVENTING_KAFKA_CONFIG_DIR}/*yaml "${EVENTING_KAFKA_CONFIG_TEMP_DIR}"
+
+  # Update The Kafka Secret With Strimzi Kafka Cluster Brokers (No Authentication)
+  sed -i "s/brokers: RU1QVFk=/brokers: ${STRIMZI_KAFKA_CLUSTER_BROKERS_ENCODED}/" "${EVENTING_KAFKA_CONFIG_TEMP_DIR}/${EVENTING_KAFKA_SECRET_TEMPLATE}"
+  sed -i "s/namespace: RU1QVFk=/namespace: \"\"/" "${EVENTING_KAFKA_CONFIG_TEMP_DIR}/${EVENTING_KAFKA_SECRET_TEMPLATE}"
+  sed -i "s/password: RU1QVFk=/password: \"\"/" "${EVENTING_KAFKA_CONFIG_TEMP_DIR}/${EVENTING_KAFKA_SECRET_TEMPLATE}"
+  sed -i "s/username: RU1QVFk=/username: \"\"/" "${EVENTING_KAFKA_CONFIG_TEMP_DIR}/${EVENTING_KAFKA_SECRET_TEMPLATE}"
+
```

Comment:
Maybe (given the configmap is now TLS/SASL disabled) we should also change the "default" secret to just be empty strings instead of TODOs - then we could also remove this bit.   Thoughts?

Reply:
Always good to simplify it.  I can't see any practical difference between using base64 "EMPTY" vs the literal empty string for those.  I left "brokers" the way it is since that needs replacing either way (unless we want to change it to "my-cluster-kafka-bootstrap.kafka.svc:90" which seems somewhat awkward to have as the default).